modules/ansible/ansibleee.go

```diff
@@ -0,0 +1,321 @@
+		setRunnerEnvVar(h, "RUNNER_EXTRA_VARS", parsedExtraVars, "extraVars", job, hashes)
+	}
+
+	hashPodSpec(h, podSpec, hashes)
+
+	_, errorHash := hashOfInputHashes(hashes)
+	if errorHash != nil {
+		return nil, fmt.Errorf("error generating hash of input hashes: %w", errorHash)
+	}
+
+	return job, nil
+}
+
+// labelsForOpenStackAnsibleEE returns the labels for selecting the resources
+// belonging to the given openstackansibleee CR name.
+func labelsForOpenStackAnsibleEE(name string, labels map[string]string) map[string]string {
+	ls := map[string]string{
+		"app":                   "openstackansibleee",
+		"job-name":              name,
+		"openstackansibleee_cr": name,
+		"osaee":                 "true",
+	}
+	for key, val := range labels {
+		ls[key] = val
+	}
+	return ls
+}
+
+func (a *EEJob) addEnvFrom(job *batchv1.Job) {
+	job.Spec.Template.Spec.Containers[0].EnvFrom = []corev1.EnvFromSource{
+		{
+			ConfigMapRef: &corev1.ConfigMapEnvSource{
+				LocalObjectReference: corev1.LocalObjectReference{Name: a.EnvConfigMapName},
+			},
+		},
+	}
+}
+
+func (a *EEJob) addMounts(job *batchv1.Job) {
+	var volumeMounts []corev1.VolumeMount
+	var volumes []corev1.Volume
+
+	// ExtraMounts propagation: for each volume defined in the top-level CR
+	// the propagation function provided by lib-common/modules/storage is
+	// called, and the resulting corev1.Volumes and corev1.Mounts are added
+	// to the main list defined by the ansible operator
+	for _, exv := range a.ExtraMounts {
+		for _, vol := range exv.Propagate([]storage.PropagationType{storage.Compute}) {
+			volumes = append(volumes, vol.Volumes...)
+			volumeMounts = append(volumeMounts, vol.Mounts...)
+		}
+	}
+
+	job.Spec.Template.Spec.Containers[0].VolumeMounts = volumeMounts
+	job.Spec.Template.Spec.Volumes = volumes
+}
+
+func hashPodSpec(
+	h *helper.Helper,
+	podSpec corev1.PodSpec,
+	hashes map[string]string,
+) {
+	var err error
+	spec, _ := podSpec.Marshal()
+	hashes["podspec"], err = calculateHash(string(spec))
+	if err != nil {
+		h.GetLogger().Error(err, "Error calculating the PodSpec hash")
+	}
+}
+
+// set value of runner environment variable and compute the hash
+func setRunnerEnvVar(
+	helper *helper.Helper,
+	varName string,
+	varValue string,
+	hashType string,
+	job *batchv1.Job,
+	hashes map[string]string,
+) {
+	var envVar corev1.EnvVar
+	var err error
+	envVar.Name = varName
+	envVar.Value = "\n" + varValue + "\n\n"
+	job.Spec.Template.Spec.Containers[0].Env = append(job.Spec.Template.Spec.Containers[0].Env, envVar)
+	hashes[hashType], err = calculateHash(varValue)
+	if err != nil {
+		helper.GetLogger().Error(err, "Error calculating the hash")
+	}
+}
+
+func calculateHash(envVar string) (string, error) {
+	hash, err := util.ObjectHash(envVar)
+	if err != nil {
+		return "", err
+	}
+	return hash, nil
+}
+
+func hashOfInputHashes(hashes map[string]string) (string, error) {
+	var err error
+	var hash string
+	var builder strings.Builder
+	for key, value := range hashes {
```

Comment:
is this a one time calc of hashOfInputHashes? otherwise it should be sorted by keys. as there is no sorted order when look over hashmap, the result can have different order when rerunning and therefore different hash sum.

Reply:
I don't think we're currently doing anything with this. I think I'll just remove this function entirely.